cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1565,6 +1564,103 @@ data MultiSigScript = RequireSignature (Hash PaymentKey)
                     | RequireMOf Int [MultiSigScript]
   deriving (Eq, Show)
 
+instance ToJSON MultiSigScript where
+  toJSON (RequireSignature payKeyHash) = String . Text.decodeUtf8 $ serialiseToRawBytesHex $ payKeyHash
+  toJSON (RequireAnyOf reqSigs) = object [ "any" .= map toJSON reqSigs ]
+  toJSON (RequireAllOf reqSigs) = object [ "all" .= map toJSON reqSigs ]
+  toJSON (RequireMOf reqNum reqSigs) =
+    case reqNum > length reqSigs of
+      False -> object [convertMofN .= map toJSON reqSigs]
+      True -> error $ "mofn: Required number of signatures \
+                      \exceed number of payment key hashes provided. m: "
+                      ++ show reqNum ++ " n: " ++ show (length reqSigs)
+   where
+     convertMofN = Text.pack $ (show reqNum ++ "of" ++ show (length reqSigs))
+
+instance FromJSON MultiSigScript where
+  parseJSON = Aeson.withObject "MultiSigScript" $ \obj -> do
+    -- Parse "all" multisig objects
+    allReqHashes <- case HMS.lookup "all" obj of
+                       Just (Array vecAllReqHashes) ->
+                         return . Just . RequireAllOf $ gatherSignatures vecAllReqHashes
+                       _ -> return Nothing
+
+    -- Parse "any" multisig objects
+    anyReqHashes <- case HMS.lookup "any" obj of
+                       Just (Array vecAnyReqHashes) ->
+                         return . Just . RequireAnyOf $ gatherSignatures vecAnyReqHashes
+                       _ -> return Nothing
+
+    -- Parse "mofn" multisig objects
+    mofN <- return $ filterMofN obj
```

Comment:
It seems inconvenient to have the client parse `m` and `n` from the JSON key. I think it'd be easier for the typical client if we stick with a static known key, such as `mOfN`, as opposed to dynamic keys, such as `3Of5` and `2Of6`.

I wonder if something like this would be more appropriate:

```json
{
  "mOfN": {
    "m": 2,
    "n": 6,
    "keyHashes": ["examplepaymentkeyhash", "examplepaymentkeyhash", "examplepaymentkeyhash", "examplepaymentkeyhash", "examplepaymentkeyhash", "examplepaymentkeyhash"]
  }
}
```

Reply:
Opinions? @erikd @newhoggy @dcoutts

Reply:
The above suggestion would be easier to parse from something like `jq`.

Reply:
Changed to:
```
{
    "atLeast": {
        "paymentKeyHashes": [
            "2f3d4cf10d0471a1db9f2d2907de867968c27bca6272f062cd1c2413",
            "f856c0c5839bab22673747d53f1ae9eed84afafb085f086e8e988614",
            "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538",
            "686024aecb5884d73a11b9ae4e63931112ba737e878d74638b78513a"
        ],
        "required": 2,
        "total": 4
    }
}
```